tools/config/__init__.py

```diff
@@ -610,24 +610,62 @@ def _get_mem_specs(self, memories, cmsis_part, exception_text):
                 continue
         raise ConfigException(exception_text)
 
-    @property
-    def rom(self):
-        """Get rom information as a pair of start_addr, size"""
+    def get_region_list(self, region='ROM'):
+        # Get list of all available regions based on memory selected in the form of list.
+        if region == 'ROM':
+            return ['IROM1', 'PROGRAM_FLASH', 'IROM2']
+        else:
+            return ['IRAM1', 'IRAM2', 'IRAM3', 'IRAM4', 'SRAM_OC', \
+                    'SRAM_ITC', 'SRAM_DTC', 'SRAM_UPPER', 'SRAM_LOWER']
+
+    def get_all_active_memories(self, *memory_list):
```

Comment:
You probably don't want to pass this in as all positional arguments. You should remove the `*`.